pubsub/pubsub.go

```diff
@@ -22,16 +22,25 @@ import (
 
 // PubSub is the interface for message buses.
 type PubSub interface {
+	Batcher
 	Init(metadata Metadata) error
 	Features() []Feature
 	Publish(req *PublishRequest) error
 	Subscribe(ctx context.Context, req SubscribeRequest, handler Handler) error
 	Close() error
 }
 
+type Batcher interface {
+	BatchPublish(req *BatchPublishRequest) BatchPublishResponse
+	BatchSubscribe(ctx context.Context, req SubscribeRequest, handler BatchHandler) error
+}
+
 // Handler is the handler used to invoke the app handler.
 type Handler func(ctx context.Context, msg *NewMessage) error
 
+// BatchHandler is the handler used to invoke the app handler.
+type BatchHandler func(ctx context.Context, msg *NewBatchMessage) error
```

Comment:
Should `BatchHandler` also return more data with error?

Reply:
Yeah, it will return.